price_feeds_poller.py

```diff
@@ -342,12 +397,36 @@ def main(args):
         w3.middleware_onion.add(middleware.simple_cache_middleware)
 
         network_gas_price = None
+        print("Gas price strategy: estimate_medium")
+      
+      elif network_gas_price is None:
+        w3.eth.set_gas_price_strategy(rpc_gas_price_strategy)
+        print("Gas price strategy: eth_gasPrice")
+      
       else:
         if network_gas_price == "estimate_medium" and w3.eth.chainId != 1:
-          print(f"Invalid gas price: {network_gas_price}. \"estimate_medium\" can only be used for mainnet (current id: {w3.eth.chainId})")
+         print(f"Invalid gas price: {network_gas_price}. \"estimate_medium\" can only be used for mainnet (current id: {w3.eth.chainId})")
         else:
-          print(f"Invalid gas price: {network_gas_price}. `gas_price` can only be an integer or \"estimate_medium\".")
+          print(f"Invalid gas price: {network_gas_price}.")
         exit(1)
+    else:    
+      print(f"Gas price strategy: invariant ({'{:,}'.format(network_gas_price)})")
+
+    # Connect to the Web3 provider
+    try:
+      current_block = w3.eth.blockNumber
+      print(f"Connected to '{network_name}' at block #{current_block} via {network_provider}")      
+
+    except Exception as ex:
+      print(f"Fatal: connection failed to {network_provider}: {ex}")
+      exit(1)
+
+    # Log Web3 client version
+    try:
+      print(f"Web3 client: {w3.clientVersion}")
+    except:
```

Comment:
I try to avoid `except:` because it also catches `KeyboardInterrupt` exceptions, making it hard to stop the process using ctrl-c.

```suggestion
    except Exception as ex:
```